_examples/dupsteps/dupsteps_test.go

```diff
@@ -0,0 +1,113 @@
+package dupsteps
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cucumber/godog"
+)
+
+func TestDuplicateSteps(t *testing.T) {
+
+	featureContents := `
+Feature:	check this out
+
+	Scenario: Flat Tire
+		Given I ran over a nail and got a flat tire
+		Then I fixed it
+		Then I can continue on my way
+
+	Scenario: Clogged Drain
+		Given I accidentally poured concrete down my drain and clogged the sewer line
+		Then I fixed it
+		Then I can once again use my sink
+`
+
+	suite := godog.TestSuite{
+		Name: t.Name(),
+		ScenarioInitializer: func(context *godog.ScenarioContext) {
+			// NOTE: loading implementations of steps for different scenarios here
+			(&cloggedDrain{}).addCloggedDrainSteps(context)
```

Comment:
The "clogged drain" scenario (above) works fine until the "flat tire" scenario (below) is added (e.g., in a subsequent commit).

The _"I fixed it"_ step from the newly added `flatTire` scenario silently overrides the matching step in the previously defined step from the `cloggedDrain` scenario.  The writer of the clogged drain scenario will have to investigate why his/her scenario suddenly stops working.

We'd like the tool to support a distributed development workflow where developers can work on scenarios independently, without having to take into account (and reuse) all pre-existing, matching steps.